### ISSUE TITLE
Python 3.10 & PIP are broken on Ubuntu

### ISSUE THREAD

**Issue by e2489** [DIGITALCRIMINALS/OnlyFans#168]
I can't speak for Windows, but Python 3.10 is in a very buggy state on Ubuntu. PIP will barely install and doesn't function correctly. Constant errors.
Had zero issues with 3.8, 3.9, etc., not sure why the 3.10 requirement was pushed so early, but those of us on Ubuntu are out of luck for now.

**Comment by aboredpervert** [comment]
Without more information it's very hard to figure out if the problem is on your side or not.

1. What Ubuntu version?
2. Are you using the system Python or the one from deadsnakes?
3. Are you invoking the "right" pip/python? (using a venv may help there)

A step-by-step dump of what you did to get there would help.

**Comment by laynlownc** [comment]
So I can speak to this issue. I've done a clean install (ubuntu-20.04.3-live-server-amd64.iso) and after an apt update and apt upgrade, I try to run
`sudo apt update && sudo apt -y install python3.10`
from 
https://github.com/DIGITALCRIMINALS/OnlyFans/blob/master/Linux.md#Installation-of-Python

I hit the following errors. 
`Hit:1 http://us.archive.ubuntu.com/ubuntu focal InRelease
Get:2 http://us.archive.ubuntu.com/ubuntu focal-updates InRelease [114 kB]
Get:3 http://us.archive.ubuntu.com/ubuntu focal-backports InRelease [108 kB]
Get:4 http://us.archive.ubuntu.com/ubuntu focal-security InRelease [114 kB]
Fetched 336 kB in 1s (633 kB/s)   
Reading package lists... Done
Building dependency tree       
Reading state information... Done
All packages are up to date.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package python3.10
E: Couldn't find any package by glob 'python3.10'`

**Comment by aboredpervert** [comment]
On Ubuntu if you want a version of Python that is not the system one you have to install through deadsnakes:

    apt install software-properties-common
    add-apt-repository ppa:deadsnakes/ppa
    apt install python3.10

**Comment by laynlownc** [comment]
Thanks for the help @aboredpervert 

For anyone else I replaced step 1 of [Installation of Python](https://github.com/DIGITALCRIMINALS/OnlyFans/blob/master/Linux.md#Installation-of-Python) with deadsnakes install above. Had to add sudo to the beginning of the lines to elevate permissions. 

When I follow that path, that is when I run into the same errors as [#177](https://github.com/DIGITALCRIMINALS/OnlyFans/issues/177) but your recommendation to Step 4 on [#43](https://github.com/DIGITALCRIMINALS/OnlyFans/issues/43) I don't think applies. Any further guidance?

**Comment by laynlownc** [comment]
To add to my information, I went and tried [Pull Request 174 ](https://github.com/DIGITALCRIMINALS/OnlyFans/pull/174) from Resokou and gett the following error

```
Traceback (most recent call last):
  File "/home/useracct/onlyfans/start_ofd.py", line 107, in <module>
    asyncio.run(main())
  File "/usr/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.10/asyncio/base_events.py", line 641, in run_until_complete
    return future.result()
  File "/home/useracct/onlyfans/start_ofd.py", line 89, in main
    api = await main_datascraper.start_datascraper(
  File "/home/useracct/onlyfans/datascraper/main_datascraper.py", line 63, in start_datascraper
    module.assign_vars(
  File "/home/useracct/onlyfans/modules/onlyfans.py", line 61, in assign_vars
    profile_directory = main_helper.get_directory(
  File "/home/useracct/onlyfans/helpers/main_helper.py", line 761, in get_directory
    directory = check_space(new_directories, min_size=min_drive_space)
  File "/home/useracct/onlyfans/helpers/main_helper.py", line 774, in check_space
    obj_Disk = disk_usage(download_path)
  File "/home/useracct/onlyfans/helpers/main_helper.py", line 78, in disk_usage
    raise FileNotFoundError(errno.ENOENT, os.strerror(errno.ENOENT), path)
FileNotFoundError: [Errno 2] No such file or directory: '/home/useracct/onlyfans/.profiles'
```

**Comment by aboredpervert** [comment]
What I said in #177 is specific to Windows.
It's possible that #174 is bugged also. I will start an Ubuntu 20 container and try it.

**Comment by aboredpervert** [comment]
After testing, it seems that `psutil` also needs to be compiled on Ubuntu.

You can install the required packages with:

    apt install gcc python3.10-dev

After doing this, the requirements installed just fine and I was able to run the script.

**Comment by laynlownc** [comment]
Thanks again for the help @aboredpervert, I was able to get up and running.

**Comment by DIGITALCRIMINALS** [comment]
Various bugs should be fixed in Python 3.10.1+
